tests/integration/redis-cli.tcl

```diff
@@ -809,3 +809,27 @@ if {!$::tls} { ;# fake_redis_node doesn't support TLS
         assert_equal "a\n1\nb\n2\nc\n3" [exec {*}$cmdline ZRANGE new_zset 0 -1 WITHSCORES]
     }
 }
+
+start_server {tags {"cli external:skip"}} {
+    test_interactive_cli_with_prompt "db_num showed in redis-cli after reconnected" {
+        run_command $fd "select 0\x0D"
+        run_command $fd "set a zoo-0\x0D"
+        run_command $fd "select 6\x0D"
+        run_command $fd "set a zoo-6\x0D"
+        r save
+
+        # kill server and restart
+        exec kill [s process_id]
+        wait_for_log_messages 0 {"*Redis is now ready to exit*"} 0 1000 10
+        catch {[run_command $fd "ping\x0D"]} err
+        restart_server 0 true false 0
+
+        # redis-cli should show '[6]' after reconnected and return 'zoo-6'
+        write_cli $fd "GET a\x0D"
+        after 100
+        set result [format_output [read_cli $fd]]
```

Comment:
the reason is the retry num(5) is too low?
maybe we can just increment the threshold?

Reply:
either way is ok for me. thanks.